python/ray/serve/master.py

```diff
@@ -219,20 +219,30 @@ def get_all_endpoints(self):
         return expand(
             self.route_table.list_service(include_headless=True).values())
 
+    def get_all_routes(self):
+        return expand(self.route_table.list_service().keys())
+
+    def get_all_backends(self):
+        return self.backend_table.list_backends()
+
     async def set_traffic(self, endpoint_name, traffic_policy_dictionary):
-        assert endpoint_name in expand(
-            self.route_table.list_service(include_headless=True).values())
+        assert endpoint_name in self.get_all_endpoints(), \
+            f"Attempted to assign traffic for an endpoint '{endpoint_name}'" \
+            " that is not registered."
 
         assert isinstance(traffic_policy_dictionary,
                           dict), "Traffic policy must be dictionary"
         prob = 0
+        existing_backends = set(self.get_all_backends())
         for backend, weight in traffic_policy_dictionary.items():
             prob += weight
-            assert (backend in self.backend_table.list_backends()
-                    ), "backend {} is not registered".format(backend)
-        assert np.isclose(
-            prob, 1, atol=0.02
-        ), "weights must sum to 1, currently it sums to {}".format(prob)
+            assert backend in existing_backends, \
+                f"Attempted to assign traffic to a backend '{backend}' that " \
+                "is not registered"
+
+            assert np.isclose(
```

Comment:
Should this be de-dented? Prob should only sum up to one after we add up all the backends right?

Reply:
Yes, it should, thanks. I didn't mean to indent it, and I've undone that.